aten/src/ATen/cpu/vec/vec512/vec512_base.h

```diff
@@ -0,0 +1,869 @@
+#pragma once
+
+// DO NOT DEFINE STATIC DATA IN THIS HEADER!
+// See Note [Do not compile initializers with AVX]

```

Comment:
This file is substantially copied from vec256_base.h without any changes besides the sizeof changes; I wonder if we can template our way to victory here

Reply:
Thanks. I'll take a look at this file. Maybe we would be able to use different alignment with `CPU_CAPABILITY` flags in a unified file.

Reply:
I'll unify them in a couple of hours. Thanks!

Reply:
`vec_base.h` & `intrinsics.h` are now unified. `SumKernel.cpp` also doesn't have duplicate code now.
However, instead of using templates, I used macros. It'd be great to have your opinion on whether this approach is okay -
https://github.com/imaginary-person/pytorch-1/blob/only_vec/aten/src/ATen/cpu/vec/vec_base.h#L35-L66.

Thanks!

Reply:
@ezyang, I only introduced 7 new macros (which could also have been `typedef`s). The `align` ones were already there.

BTW, thank you for your patience with the graphs!
To avoid any further delay, I'd dedicatedly work on them over the weekend.
Thanks!

Reply:
BTW, @ezyang, ResNext101 doesn't benefit with AVX512 support in ATen. I'll profile it over the weekend to check why, but my best guess is that it heavily uses FBGEMM, which already has AVX512 support.